File: others/0035-SearchInsertPosition/Solution.java
public class SearchInsertPosition {
    public static int searchInsertPosition(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] >= target) {
                return i;
            }
        }
        return nums.length;
    }

    public static void main(String[] args) {
        int[] array = new int[]{1, 3, 5, 6};
        System.out.println("expected: 2, actual: " + searchInsertPosition(array, 5));
        System.out.println("expected: 1, actual: " + searchInsertPosition(array, 2));
        System.out.println("expected: 4, actual: " + searchInsertPosition(array, 7));
        System.out.println("expected: 0, actual: " + searchInsertPosition(array, 0));
    }

}

File: others/0035-SearchInsertPosition/solution.py
class Solution:

    def searchInsert(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums) - 1
        while i <= j:
            m = (i + j) // 2
            n = nums[m]
            if target < n:
                j = m - 1
            elif target > n:
                i = m + 1
            else:  # n == target
                return m
        return i


if __name__ == "__main__":
    s = Solution()

